src/deluge/storage/cluster/cluster.h

```diff
@@ -29,25 +32,51 @@ class SampleCache;
 
 class Cluster final : public Stealable {
 public:
-	Cluster();
+	constexpr static size_t kSizeFAT16Max = 32768;
+	enum class Type {
+		EMPTY,
+		SAMPLE,
+		GENERAL_MEMORY,
+		SAMPLE_CACHE,
+		PERC_CACHE_FORWARDS,
+		PERC_CACHE_REVERSED,
+		OTHER,
+	};
+
+	static Cluster* alloc(Cluster::Type type = Cluster::Type::SAMPLE, bool shouldAddReasons = true,
+	                      void* dontStealFromThing = nullptr);
+
+	static void dealloc(Cluster& cluster);
```

Comment:
Could this be a non-static method operating on `this` instead? It feels silly to need to call it using `Cluster` when 95% of the time the variable name is `cluster` anyway (and the type information for the variable is easy to find).

Reply:
Mmmmmaybe? It's generally seen as bad practice to call a destructor on `this` from inside a member function, just because of all the caveats. https://isocpp.org/wiki/faq/freestore-mgmt#delete-this

Reply:
Eh, yeah I can do that.